File: cpp/PartTimeEmployee.cpp
#include "PartTimeEmployee.h"

PartTimeEmployee::PartTimeEmployee(const std::string& name, int id, float hourlyRate, int hoursWorked)
    : Employee(name, id), hourlyRate(hourlyRate), hoursWorked(hoursWorked) {}

float PartTimeEmployee::calculateSalary() const {
    return hourlyRate * hoursWorked;
}

File: src/models/PartTimeEmployee.ts
import {Employee} from "../interface/IEmployee";

class PartTimeEmployee extends Employee {
     hourlyRate: number;
     hoursWorked: number;

    constructor(name: string, id: number, hourlyRate: number, hoursWorked: number) {
        super(name, id);
        this.hourlyRate = hourlyRate;
        this.hoursWorked = hoursWorked;
    }

    calculateSalary(): number {
        return this.hourlyRate * this.hoursWorked;
    }
}

export {PartTimeEmployee};
